templates/kotlin/src/main/kotlin/io/appwrite/extensions/TypeExtensions.kt.twig

```diff
@@ -1,8 +1,107 @@
 package {{ sdk.namespace | caseDot }}.extensions
 
+import {{ sdk.namespace | caseDot }}.models.Payload
 import kotlin.reflect.KClass
 import kotlin.reflect.typeOf
 
 inline fun <reified T : Any> classOf(): Class<T> {
     return (typeOf<T>().classifier!! as KClass<T>).java
-}
\ No newline at end of file
+}
+
+fun String.fromMultiPart(binaryBody: ByteArray): Map<String, Any> {
+    val match = Regex("(-+\\w+)--").find(this) ?: return emptyMap()
```

Comment:
Is regex slow? Can we use a faster check here?

Reply:
In ideal world we would use well tested library for this, but to release quickly, we use this snippet.

Stability for us should be fine, as tests pass.

Speed for us should be fine, we only have a few keys. Could be a bit slower for 20MB payload, but regex looks pretty simple so I wouldn't be too worried